pkg/operator/staticpod/installerpod/cmd.go

```diff
@@ -438,6 +446,130 @@ func (o *InstallOptions) Run(ctx context.Context) error {
 	return nil
 }
 
+func (o *InstallOptions) waitForOtherInstallerRevisionsToSettle(ctx context.Context) error {
+	currRevision64, err := strconv.ParseInt(o.Revision, 10, 32)
+	if err != nil {
+		return fmt.Errorf("bad local revision %v: %w", o.Revision, err)
+	}
+	currRevision := int(currRevision64)
+
+	// at this point we have a list of all the installer pods present on this node we need to check
+	// 1. is there a revision later than us present right now.  If yes, exit.
+	// 2. is there an installer pod present that has not finished running, wait until it is finished.
+	err = wait.PollImmediateWithContext(ctx, 10*time.Second, 60*time.Second, func(ctx context.Context) (done bool, err error) {
+		installerPods, err := o.getInstallerPodsOnThisNode(ctx)
+		if err != nil {
+			return false, nil
+		}
+		if len(installerPods) == 0 {
+			return false, fmt.Errorf("no installer pods found")
+		}
+		latestRevision, err := getRevisionOfPod(installerPods[len(installerPods)-1])
+		if err != nil {
+			return false, err
+		}
+		if latestRevision > currRevision {
```

Comment:
Assuming an installer pod with lower revision does not get created after an installer pod with higher revision, lower revision installer pod will never get through this point.

This condition might cause the installation for a given master node to get stuck. Assuming there are installer rev-6 and rev-7 created, rev-6 gets to run before rev-7 (the usual case) but the rev-7 for some reason never starts (e.g. kubelet failing to create a sandbox or other dependent components like secrets). Now the installer rev-6 never gets through this, the installer rev-7 never starts and the installed static pod never gets updated. The installer controller will hang and the operator goes into Degraded=true state.

Reply:
rephrasing the timeline to be sure I understand it.
1. installer-6 created
2. installer-7 created
3. installer-6 running
4. installer-6 exits because it's not the latest (this is good)
5. kubelet never starts installer-7 for some reason
6. operator goes degraded.

I think this situation is expected and ok.